app/src/main/java/com/allan/boardbuddies/NoteAdapter.java

```diff
@@ -39,12 +43,24 @@ public int getItemCount() {
     }
 
     // Allows access to each list item view without needing to lookup every time
-    public static class NoteViewHolder extends RecyclerView.ViewHolder{
+    public static class NoteViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
         TextView titleTextView;
+        OnNoteListener onNoteListener;
 
-        public NoteViewHolder(View itemView){
+        public NoteViewHolder(View itemView, OnNoteListener onNoteListener){
             super(itemView);
             titleTextView = itemView.findViewById(R.id.note_title_text);
+            itemView.setOnClickListener(this);
```

Comment:
You don't need to make your viewholder be the OnClickListener. Instead, just do this

itemView.setOnClickListener(new View.OnClickListener() {
  @Override
  public void onClick(View v) {
    this.OnNoteListener.onNoteClick(getBindingAdapterPosition());
  }
}